### PR TITLE
StorageDriver: GCS: remove support for directory Moves

### PR DESCRIPTION
**Pull request by aibaars**
The Move operation is only used to move uploaded blobs to their final destination. There is no point in implementing Move on "folders". Apart from simplifying the code, this also saves an HTTP request in the GCS driver.

### CODE REVIEW

**Comment by aibaars** [comment]
Closing this for now. Will rebase and re-open when #1395 has been implemented.

**Comment by stevvooe** [comment]
@aibaars Could you add a testsuite entry for this? Calling `Move` with a directory source should return an error.

**Comment by aibaars** [comment]
@stevvooe
I added a test case to the GCS driver tests that ensures that the driver returns an error in case of a directory move.

I do not think this should be a test in the test suite for all storage drivers, because I think it is OK for other storage drivers to implement directory moves (filesystem and inmemory, and  possibly some others do). It is just a feature that is never used.

File: registry/storage/driver/gcs/gcs.go [hunk 1]
```diff
@@ -514,7 +479,10 @@ func (d *driver) Move(context ctx.Context, sourcePath string, destPath string) e
 		}
 		return err
 	}
-	return storageDeleteObject(gcsContext, d.bucket, d.pathToKey(sourcePath))
+	// if deleting the file fails, ignore the error; the file was succesfully copied,
+	// and the original should eventually be cleaned when purging the uploads folder
+	storageDeleteObject(gcsContext, d.bucket, d.pathToKey(sourcePath))
```
Comment (RichardScothern) [inline registry/storage/driver/gcs/gcs.go]: Let's  add an info log for this case

**Comment by RichardScothern** [comment]
One nit, but LGTM

**Comment by stevvooe** [comment]
LGTM

File: registry/storage/driver/gcs/gcs.go (hunk 1 above)
Comment (aibaars) [inline registry/storage/driver/gcs/gcs.go]: Done!

**Pull request merged**